### PR TITLE
Add preview of generated Legends

### PR DESCRIPTION
**Pull request by RobertDenis**
Adding a legend checkbox and backend call to the compose map screen.

### CODE REVIEW

File: geoserver/webapp/app/core/backend.js [hunk 1]
```diff
@@ -378,6 +378,15 @@ angular.module('gsApp.core.backend',[])
             }
           },
 
+          //legend graphic
+          legend: {
+            get: function(layer) {
+              var url = gsRoot + '/wms?REQUEST=GetLegendGraphic&VERSION=1.0&'+
+                'FORMAT=image/png&WIDTH=20&HEIGHT=20&LAYER='+layer;
```
Comment (jodygarnett) [inline geoserver/webapp/app/core/backend.js]: width and height is a bit odd for legend graphics (for styles with several rules you may get back a different size then you expect.

See: http://docs.geoserver.org/latest/en/user/services/wms/get_legend_graphic/legendgraphic.html

Do you have any need of adjusting the DPI setting or other options mentioned on the above page?

**Comment by eshon** [comment]
Is this the expected output?  I think it's confusing in this state.  Will not merge, please review with Jody who requested this feature what the ideal output for a Legend should be.

screen shot 2015-02-09 at 3 01 18 pm
screen shot 2015-02-09 at 3 01 01 pm

**Comment by jodygarnett** [comment]
As long as the legend is responding to current zoom the above screen snaps look correct.

The last example mostly looks bad as the title / description has not been filled in.

**Comment by eshon** [comment]
Why is the first legend only showing streams?

**Comment by jodygarnett** [comment]
I assume that is the layer they have selected? And thus the style they want previewed.

**Comment by eshon** [comment]
That's not clear when placing that checkbox in the UI like that.  I will reassign ticket in JIRA to myself to think about this one.

**Pull request closed without merging**